solr/core/src/test/org/apache/solr/cloud/TestTlogReplica.java

```diff
@@ -576,43 +576,31 @@ public void testOnlyLeaderIndexes() throws Exception {
 
     {
       SolrCore core = getSolrCore(true).getFirst();
-      var reader =
-          core.getSolrMetricsContext()
-              .getMetricManager()
-              .getPrometheusMetricReader(
-                  getSolrCore(true).getFirst().getCoreMetricManager().getRegistryName());
       var actual =
-          (GaugeSnapshot.GaugeDataPointSnapshot)
-              SolrMetricTestUtils.getDataPointSnapshot(
-                  reader,
-                  "solr_metrics_core_update_pending_operations",
-                  SolrMetricTestUtils.getCloudLabelsBase(core)
-                      .get()
-                      .label("category", "UPDATE")
-                      .label("operation", "docs_pending")
-                      .build());
+          SolrMetricTestUtils.getGaugeDatapoint(
+              core,
+              "solr_core_update_docs_pending_commit",
+              Labels.builder()
+                  .label("category", "UPDATE")
+                  .label("operation", "docs_pending")
+                  .build(),
+              true);
       assertEquals(
           "Expected 4 docs are pending in core " + getSolrCore(true).get(0).getCoreDescriptor(),
           4,
           (long) actual.getValue());
     }
 
     for (SolrCore solrCore : getSolrCore(false)) {
-      var reader =
-          solrCore
-              .getSolrMetricsContext()
-              .getMetricManager()
-              .getPrometheusMetricReader(solrCore.getCoreMetricManager().getRegistryName());
       var actual =
-          (GaugeSnapshot.GaugeDataPointSnapshot)
-              SolrMetricTestUtils.getDataPointSnapshot(
-                  reader,
-                  "solr_metrics_core_update_pending_operations",
-                  SolrMetricTestUtils.getCloudLabelsBase(solrCore)
-                      .get()
-                      .label("category", "UPDATE")
-                      .label("operation", "docs_pending")
-                      .build());
+          SolrMetricTestUtils.getGaugeDatapoint(
+              solrCore,
+              "solr_core_update_docs_pending_commit",
+              Labels.builder()
+                  .label("category", "UPDATE")
+                  .label("operation", "docs_pending")
+                  .build(),
+              true);
```

Comment:
nit: booleans like here are inscrutable when reviewing code.  Not a big deal though